web/impact/impact/tests/test_allocate_applications_view.py

```diff
@@ -0,0 +1,116 @@
+            assert [JUDGING_ROUND_INACTIVE_ERROR.format(judging_round.id)
+                    in response.data]
+
+    def test_get_already_assigned(self):
+        context = JudgeFeedbackContext()
+        judging_round = context.judging_round
+        judge_id = ExpertFactory().id
+        with self.login(email=self.basic_user().email):
+            url = reverse(AllocateApplicationsView.view_name,
+                          args=[judging_round.id, judge_id])
+            response = self.client.get(url)
+            assert response.status_code == 200
+            count = len(response.data)
+            response = self.client.get(url)
+            assert response.status_code == 403
+            assert [ALREADY_ASSIGNED_ERROR.format(judge=context.judge.email,
+                                                  count=count)
+                    in response.data]
+
+    def test_get_no_app_for_judge(self):
+        context = JudgeFeedbackContext()
+        judge = ExpertFactory()
+        for app in context.applications:
+            JudgeApplicationFeedbackFactory(panel=context.panel,
+                                            judge=judge,
+                                            application=app)
+        with self.login(email=self.basic_user().email):
+            url = reverse(AllocateApplicationsView.view_name,
+                          args=[context.judging_round.id, judge.id])
+            response = self.client.get(url)
+            assert response.status_code == 403
+            assert [NO_APP_LEFT_FOR_JUDGE.format(context.judge.email)
+                    in response.data]
+
+    def test_get_adds_capacity_and_quota(self):
+        context = JudgeFeedbackContext()
+        judging_round = context.judging_round
+        judge = ExpertFactory()
+        commitment = JudgeRoundCommitmentFactory(judge=judge,
+                                                 judging_round=judging_round,
+                                                 capacity=0,
+                                                 current_quota=0)
+        with self.login(email=self.basic_user().email):
+            url = reverse(AllocateApplicationsView.view_name,
+                          args=[judging_round.id, judge.id])
+            response = self.client.get(url)
+            assert response.status_code == 200
+            commitment.refresh_from_db()
+            assert commitment.capacity > 0
+            assert commitment.current_quota > 0
+
+    def test_get_by_judge_succeeds(self):
+        context = JudgeFeedbackContext()
+        judging_round = context.judging_round
+        judge = ExpertFactory()
+        judge.set_password("password")
+        judge.save()
+        with self.login(email=judge.email):
+            url = reverse(AllocateApplicationsView.view_name,
+                          args=[judging_round.id, judge.id])
+            response = self.client.get(url)
+            assert response.status_code == 200
```

Comment:
Anything we can test about the content? status_code == 200 is a pretty wide net.

Reply:
Pushing back.  All I'm testing is that the judge can access the page so this seems like the right test to me.  We are testing content elsewhere.  Exactly what it means to run this API call as the judge vs. as a staff member could at least in theory change.  Note that this is the same check that we have for test_get.